### PR TITLE
Add postgresql-debversion

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: packages/katello/postgresql-debversion/postgresql-debversion.spec [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+%build
+make %{?_smp_mflags}
+
+%install
+rm -rf $RPM_BUILD_ROOT
+%make_install
+
+%files
+/usr/lib64/pgsql/debversion.so
```
Comment (ekohl) [inline packages/katello/postgresql-debversion/postgresql-debversion.spec]:
```suggestion
%{_libdir}/pgsql/debversion.so
```

File: packages/katello/postgresql-debversion/postgresql-debversion.spec [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+/usr/share/pgsql/extension/debversion--1.0.5--1.0.6.sql
```
Comment (ekohl) [inline packages/katello/postgresql-debversion/postgresql-debversion.spec]:
```suggestion
%{_datadir}/pgsql/extension/debversion--1.0.5--1.0.6.sql
```

File: packages/katello/postgresql-debversion/0001-Copy-relevant-code-from-apt-pkg-to-ease-packaging.patch [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
++    }
++    else
++      return 0;
++  }
++  // -------------------------------------------------------------------------
++
+   int32
+-  debversioncmp (text *left,
+-		 text *right)
++  debversioncmp (text *left, text *right)
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply packages/katello/postgresql-debversion/0001-Copy-relevant-code-from-apt-pkg-to-ease-packaging.patch]: removed...

**Comment by mdellweg** [comment]
@ekohl can you see, why it does not seem to like the patch?

**Comment by ekohl** [comment]
I don't really see why the scratch build fails. Perhaps there's a permission issue but right now I don't see it. However, I do think we should consider the bigger picture. In nightly we've switched over to rh-postgresql10-postgresql-server so the extension should be built against that. Do you plan the installer to ensure the extension is installed? I have also seen complexities when upgrading because the extension needs to be installed prior to upgrading. This will be relevant when we upgrade from rh-postgresql10 to rh-postgresql12 once it's available.

**Comment by jlsherrill** [comment]
[test rpm]

**Comment by jlsherrill** [comment]
@ekohl was this error the same as before?

```
[2020-02-05T20:49:50.459Z] TASK [setup_sources : Search annex for web link] *******************************

[2020-02-05T20:49:50.459Z] fatal: [postgresql-debversion]: FAILED! => changed=false 

[2020-02-05T20:49:50.459Z]   cmd: git annex whereis /home/jenkins/workspace/foreman-packaging-rpm-pr-test@2/packages/katello//postgresql-debversion/postgresql-debversion-v1.1.1.tar.gz 2>/dev/null | grep -q "web:"

[2020-02-05T20:49:50.459Z]   delta: '0:00:00.269157'

[2020-02-05T20:49:50.459Z]   end: '2020-02-05 20:49:50.386345'

[2020-02-05T20:49:50.459Z]   msg: non-zero return code

[2020-02-05T20:49:50.459Z]   rc: 1

[2020-02-05T20:49:50.459Z]   start: '2020-02-05 20:49:50.117188'

[2020-02-05T20:49:50.459Z]   stderr: ''

[2020-02-05T20:49:50.459Z]   stderr_lines: <omitted>

[2020-02-05T20:49:50.459Z]   stdout: ''

[2020-02-05T20:49:50.459Z]   stdout_lines: <omitted>

[2020-02-05T20:49:50.459Z] ...ignoring
```

Locally it seems fine, so i'm scratching my head

**Comment by jlsherrill** [comment]
[test rpm]

**Comment by jlsherrill** [comment]
@mdellweg i figured out the build error here: https://github.com/theforeman/foreman-packaging/pull/4741 

I can see about converting it to the scl if you want me to take it over

**Comment by mdellweg** [comment]
Closing this. #4741 has been merged...

**Pull request closed without merging**